.github/workflows/doc-build.yml

```diff
@@ -69,6 +69,7 @@ jobs:
     - name: west setup
       run: |
         west init -l .
+        west update bsim
```

Comment:
I'm not sure I follow here, how is bsim related to docs?

Reply:
It's not, but the docs are using `west` and west needs to resolve the manifest. Because we now have an `import` statement in the main manifest, we need the `west update bsim` to pull the manifest that gets `import`ed

Reply:
The key here is that, for the first time, we have an `import` statement in the default upstream manifest.

Reply:
It's because of this error:
https://github.com/zephyrproject-rtos/zephyr/pull/55696#issuecomment-1481530833
That is elaborated a bit more here:
https://github.com/zephyrproject-rtos/west/issues/644

west init does not fetch the project which has the sub-manifest, so following west commands that look thru the manifest would error out as they will not find that imported manifest). The solution is to do either a full west update after west init, or to fetch just the missing project which has that sub-manifest

Reply:
does that mean this will impact all users then? too me it looks like a workaround due to a bug, am I wrong? If that's the case please add a `FIXME`.

Reply:
I don't think this is a bug, no. `west init` works and `west update` works fine. What has changed is that you cannot execute things like `west list` before `west update`, but that has never been something we've publicized.

Reply:
The difference is that now you if you do something like `west list` before you run `west update` you will get:

```
❯ west list
FATAL ERROR: failed manifest import in bsim (tools/bsim)
  Failed importing "west.yml" from revision "bc22891208d453a8cb851e4ecdc8f165f3a33890"
  Hint: for this to work:
          - bsim must be cloned
          - its manifest-rev ref must point to a commit with the import data
        To fix, run:
          west update
```
or an exception thrown if you do it via the west API (like you saw in the doc-build workflow). But this is not a bug, this is the correct west behavior as it stands now. What @aescolar proposes in https://github.com/zephyrproject-rtos/west/issues/644 is an enhancement.

Reply:
> I don't think this is a bug, no. `west init` works and `west update` works fine. What has changed is that you cannot execute things like `west list` before `west update`, but that has never been something we've publicized.

but we're just running `west init --local .`, how is west list involved here?

Reply:
> @gmarull the issue was triggered in the documentation build cmake files which use zephyr/cmake/modules/zephyr_module.cmake

Correct, the script `zephyr_module.cmake` invokes `zephyr_module.py` which in turn invokes `west`. This is necessary to generate all the `Kconfig.modules` stuff that Kconfig needs to parse the Kconfig symbols, among other things.